happi/cli.py

```diff
@@ -94,53 +96,68 @@ def search(ctx, show_json, names, search_criteria):
                 'Received duplicate search criteria %s=%r (was %r)',
                 criteria, value, client_args[criteria]
             )
-            return
-        if value.replace('.', '').isnumeric():
-            logger.debug('Changed %s to float', value)
-            value = str(float(value))
+            raise click.Abort()
 
         if is_a_range(value):
```

Comment:
Not a required change but a refactor thought: it would be nice for reusability's sake to say "give me a CLI argument (as the user would format a key/value pair or a range) and turn it into a dictionary of query parameters"

Reply:
I thought about this briefly, and the other place we use this kind of parsing is in `happi edit`.  The similarities end around the `arg.split('=', 1)` point, so I didn't think it was worth splitting it out at that point.  But it would certainly make the search block more readable.